packages/react-reconciler/src/ReactFiberHooks.js

```diff
@@ -1256,10 +1256,21 @@ function updateReducerImpl<S, A>(
     queue.pending = null;
   }
 
-  if (baseQueue !== null) {
+  const baseState = hook.baseState;
+  if (baseQueue === null) {
+    // If there are no pending updates, then the memoized state should be the
+    // same as the base state. Currently these only diverge in the case of
+    // useOptimistic, because useOptimistic accepts a new baseState on
+    // every render.
+    if (hook.memoizedState !== baseState) {
```

Comment:
should this use is()?

or just assign unconditionally

Reply:
yeah I'll just assign unconditionally